pkg/client/client.go

```diff
@@ -50,9 +51,25 @@ type tlsBootstrapClientImpl struct {
 	resource       string
 }
 
+var newLoadExecCredential = func() (*datamodel.ExecCredential, error) {
+	return loadExecCredential()
+}
+
+var newGetTlsConfig = func(pemCAs []byte, c *tlsBootstrapClientImpl, execCredential *datamodel.ExecCredential) (*tls.Config, error) {
+	return getTLSConfig(pemCAs, c.nextProto, execCredential.Spec.Cluster.InsecureSkipTLSVerify)
+}
+
+var newLoadAzureJson = func() (*datamodel.AzureConfig, error) {
+	return loadAzureJSON()
```

Comment:
for file I/O I think we can use a similar pattern to what I used for the bootstrap server implementation in RP - can follow up offline